scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -673,7 +664,17 @@
             ;; Need to lock on fenzo when accessing scheduleOnce because scheduleOnce and
             ;; task assigner can not be called at the same time.
             ;; task assigner may be called when reconciling
+            to-unassign (util/set-atom! unassign-task-set #{})
+            ^Action2 unassigner (.getTaskUnAssigner fenzo)
             result (locking fenzo
+                     (doseq [[task-id hostname] to-unassign]
+                       (try
+                         (log/debug "In" pool-name "pool, unassigning task"
+                                    task-id "from" hostname)
+                         (.call unassigner task-id hostname)
+                         (catch Exception e
+                           (log/error e "In" pool-name "pool, failed to unassign task"
+                                      task-id "from" hostname))))
```

Comment:
Would be nice to include a comment about why Fenzo task unassignment needs to happen.